devtools/server/actors/replay/connection.js

```diff
@@ -1265,25 +1277,36 @@ async function fetchText(recordingId, url) {
   }
 
   try {
-    const response = await fetch(url, {
-      //
-      credentials: "include",
+    const channel = NetUtil.newChannel({
+      uri: urlObj.toString(),
+      loadingPrincipal: browser.contentPrincipal,
+      triggeringPrincipal: browser.contentPrincipal,
+      contentPolicyType: Ci.nsIContentPolicy.TYPE_DOCUMENT,
+      securityFlags: Ci.nsILoadInfo.SEC_ALLOW_CROSS_ORIGIN_SEC_CONTEXT_IS_NULL,
     });
-    if (response.status < 200 || response.status >= 300) {
-      console.error("Error fetching recording resource", url, response);
-      pingTelemetry("sourcemap-upload", "fetch-bad-status", {
-        message: `Request got status: ${response.status}`,
-        status: response.status,
-        url: ["http:", "https:"].includes(urlObj.protocol) ? url : urlObj.protocol,
-        recordingId,
-      });
-      return null;
-    }
 
-    return {
-      url,
-      text: await response.text(),
-    };
+    return await new Promise(resolve => {
+      NetUtil.asyncFetch(channel, function(inputStream, resultCode) {
```

Comment:
FYI looks like `asyncFetch` can automatically create the channel so we could pass the object you're passing to `newChannel` directly to `asyncFetch`: https://searchfox.org/mozilla-central/source/netwerk/base/NetUtil.jsm#144